sw/device/tests/rv_core_ibex_rnd_test.c

```diff
@@ -18,20 +19,29 @@
 // Initialize OTTF.
 OTTF_DEFINE_TEST_CONFIG();
 
-// Declare two assembly function defined in `rv_core_ibex_rnd_test.S`.
+// Declare two assembly functions defined in `rv_core_ibex_rnd_test.S`.
 extern volatile uint32_t rv_core_ibex_rnd_read_and_immediately_check_status();
 extern volatile uint32_t
 rv_core_ibex_check_rnd_read_possible_while_status_invalid();
 
-#define RANDOM_DATA_READS 32
-#define MAX_STATUS_CHECKS 1024
+enum {
+  kRandomDataReads = 32,
+  // Timeout (in microseconds) when polling random data or the validity of
+  // random data.  Can currently take up to 80 ms on the FPGA and up to 2.5 s
+  // in Verilator (in simulated time; empirically determined at the time
+  // of commit) due to the rather slow rate of emulated entropy.  In DV, the
+  // emulated entropy rate gets constrained by overriding `rng_srate_value_max`
+  // so that the overall test does not exceed the default timeout of 60 minutes.
+  kTimeoutUsec = 2500000,
```

Comment:
Note that the high timeout value in Verilator is somewhat artificial: it takes ca. 1.25 M Ibex clock cycles before the core can consume entropy after it has enabled Entropy Source's Auto mode. As SW assumes an Ibex clock frequency of 500 kHz when running in Verilator (see `sw/device/lib/arch/device_sim_verilator.c`), this translates to 2.5 s of "simulated time", so 2.5e6 us. I'm not convinced that using 500 kHz as Ibex clock frequency in Verilator is meaningful, though.

Reply:
I've created issue #18327 to track this